### PR TITLE
Issue #440: add method to make changes persistent

### PR DESCRIPTION
**Pull request by jotsetung**
This pull request adds the method to make data manipulations persistent.

The reason to add yet another method to the backend was that I did not see another (clean) way to have different options to deal with applysing data manipulations: *in-memory*  backends simply apply the manipulations, but for some *on-disk* backends it is simply not possible to first load all spectra data, apply manipulations and then save it again. See issue #440 for more details.

With that method I believe we have all principal functionality in place.

### CODE REVIEW

**Comment by jotsetung** [comment]
Agree.

**Comment by jotsetung** [comment]
Especially since we already a related discussion on where to put the processing queue. Ideally, the `backend` should be agnostic of the lazy processing queue - only, I didn't find another way to implement a memory-efficient solution for ondisk backends (that don't have to load first all spectra, apply the processing queue to then save all spectra again).

**Comment by jotsetung** [comment]
A fresh look at the code might indeed be good. Your suggestion sounds nice.

**Comment by jotsetung** [comment]
@sgibb 's changes are now in this PR. Looks better now.

File: R/BackendHdf5-class.R [hunk 1]
```diff
@@ -257,11 +257,11 @@ setMethod("backendReadSpectra", "BackendHdf5", function(object, spectraData,
     h5 <- H5Fopen(h5file)
     on.exit(invisible(H5Fclose(h5)))
     comp_level <- .hdf5_compression_level()
-    if (prune) {
-        h5delete(h5, "spectra")
-    }
-    if (!H5Lexists(h5, "spectra"))
-        h5createGroup(h5, "spectra")
+    x <- force(x)
```
Comment (sgibb) [inline R/BackendHdf5-class.R]: Do we really need `force` here? To be honest I never used it before.

File: R/MSnExperiment-class.R [hunk 2]
```diff
@@ -479,53 +496,84 @@ readMSnExperiment <- function(file, sampleData, backend = BackendMzR(),
...
+        stop("Can not change backend to 'BackendMzR' because the ",
+             "data was changed.")
+    object@backend <- backendInitialize(backend, fileNames(object),
+                                        object@spectraData, ...)
+    validObject(object)
+    object
 })
 
+#' @rdname MSnExperiment
+applyProcessingQueue <- function(x, BPPARAM = bpparam()) {
```
Comment (sgibb) [inline R/MSnExperiment-class.R]: BTW: I know we had this discussion before but is there a specific reason why this is a function and not a method?

File: R/BackendHdf5-class.R (hunk 1 above)
Comment (jotsetung) [reply R/BackendHdf5-class.R]: Yes! I had it without and got very weird errors. In fact it's R's lazy evaluation. I passed `x = spectra(object)` to the function and R was first deleting the file contents and then evaluating the `spectra(object)` expression - but the files were already empty. So, with `force` we make sure that we first read the spectra from the files **before** we delete their content.

Comment (sgibb) [reply R/BackendHdf5-class.R]: Ok, makes sense, thanks.

**Pull request merged**